tests/test_subset.py

```diff
@@ -102,11 +102,12 @@ def test_time_dates_outofbounds(self):
         yr_st = "1950"
         yr_ed = "2099"
 
-        out = subset.subset_gridpoint(
-            da,
-            start_date="{yr_st}-01".format(yr_st=yr_st),
-            end_date="{yr_ed}-01".format(yr_ed=yr_ed),
-        )
+        with pytest.warns(UserWarning):
+            out = subset.subset_gridpoint(
+                da,
+                start_date="{yr_st}-01".format(yr_st=yr_st),
```

Comment:
Since we dropped 3.5, maybe you could use f-strings here ?

Reply:
True. Will do.

Reply:
Actually, changing this will lead to Black failures. This will need to happen in a new PR, but noted.